gcc/cp/semantics.cc

```diff
@@ -12593,6 +12593,12 @@ finish_decltype_type (tree expr, bool id_expression_or_member_access_p,
       if (identifier_p (expr))
         expr = lookup_name (expr);
 
+      /* If e is a constified expression inside a contract assertion,
+	 strip the const wrapper. Per P2900R14, "For a function f with the
+	 return type T , the result name is an lvalue of type const T , decltype(r)
+	 is T , and decltype((r)) is const T&."  */
+      STRIP_ANY_CONTRACT_CONST_WRAPPER (expr);
```

Comment:
So, we could just get rid of this macro here and do
expr = strip_any_contract_const_wrapper (expr);

Reply:
Or, along the suggested naming scheme,
expr = strip_contract_constify_expr (expr);